src/Service.Host/client/src/components/Workstation.js

```diff
@@ -285,113 +273,97 @@ function Workstation({ creating }) {
             const currentRow = workStation?.workStationElements.find(
-                r => r.workStationElementId === searchDialogOpen.forRow
+                r => r.workStationElementId === storageDialogOpenForRow
             );
 
             let newRow = {
                 ...currentRow,
-                [s.field]: selected.id,
                 locationId: selected.locationId,
                 locationCode: selected.locationCode,
                 locationDescription: selected.description
             };
 
-            processRowUpdate(newRow, currentRow);
-            setSearchDialogOpen({ forRow: null, forColumn: null });
+            processRowUpdate(newRow);
+            handleClose();
         };
 
         return (
-            <div id={s.field}>
-                <Dialog open={searchDialogOpen.forColumn === s.field} onClose={handleClose}>
-                    <DialogTitle>Search</DialogTitle>
-                    <DialogContent>
-                        <Search
-                            autoFocus
-                            propertyName={`${s.field}-searchTerm`}
-                            label=""
-                            resultsInModal
-                            resultLimit={100}
-                            value={storageLocationSearchTerm}
-                            handleValueChange={(_, newVal) => setStorageLocationSearchTerm(newVal)}
-                            search={s.search}
-                            searchResults={s.searchResults?.map(r => ({
-                                ...r,
-                                id: r[s.field]
-                            }))}
-                            searchLoading={s.loading}
-                            priorityFunction="closestMatchesFirst"
-                            onResultSelect={handleSearchResultSelect}
-                            clearSearch={() => {}}
-                        />
-                    </DialogContent>
-                    <DialogActions>
-                        <Button onClick={handleClose}>Close</Button>
-                    </DialogActions>
-                </Dialog>
-            </div>
+            <Dialog open={!!storageDialogOpenForRow} onClose={handleClose}>
+                <DialogTitle>Search Storage Location</DialogTitle>
+                <DialogContent>
+                    <Search
+                        autoFocus
+                        propertyName="storageLocation-searchTerm"
+                        label=""
+                        resultsInModal
+                        resultLimit={100}
+                        value={storageLocationSearchTerm}
+                        handleValueChange={(_, newVal) => setStorageLocationSearchTerm(newVal)}
+                        search={searchStorageLocations}
+                        searchResults={storageLocationsSearchResults?.map(r => ({
+                            ...r,
+                            id: r.locationId
+                        }))}
```

Comment:
you can delete the type value from the columns array entry now too:
 type = 'search' (since search is not a supported mui DataGrid column type anyway - it'll either just ignore it or throw warnings)
 
+ also  all the redundant fields from the columns array now too can be deleted to avoid confusion, things like
 
 search:
 searchResults
 searchLoading
clearSearch

these aren't part of the datagrid columns prop api anyway, so the datagrid will just ignore them